ArduPlane/mode_takeoff.cpp

```diff
@@ -83,7 +83,10 @@ void ModeTakeoff::update()
     if (!takeoff_mode_setup) {
         plane.auto_state.takeoff_altitude_rel_cm = alt * 100;
         const uint16_t altitude = plane.relative_ground_altitude(false,true);
-        const float direction = degrees(ahrs.get_yaw());
+        const Vector2f &groundspeed2d = ahrs.groundspeed_vector();
+        const float direction = wrap_360(degrees(groundspeed2d.angle()));
+        const float groundspeed = groundspeed2d.length();
+
         // see if we will skip takeoff as already flying
         if (plane.is_flying() && (millis() - plane.started_flying_ms > 10000U) && ahrs.groundspeed() > 3) {
```

Comment:
```suggestion
        if (plane.is_flying() && (millis() - plane.started_flying_ms > 10000U) && groundspeed > 3) {
```

Reply:
If there a chance that the ground speed check here will trigger before the new one on line 129. If your min airspeed is larger than 10m/s (3/0.3) and it takes longer than 10 seconds to get there then its will skip the full takeoff every time.  Really this skip takeoff path should only be done once. Maybe it should be moved to the mode enter?